### PR TITLE
Fix some issues with Docker for Windows support

### PR DESCRIPTION
**Pull request by damienalexandre**
Tried on PowerShell Core + Windows 10 I has some issues.

- python 2.7.17 is broken, we should avoid it on Windows cf https://github.com/pypa/pipenv/issues/4016
- colorama is needed under Windows because it's not required automatically: https://github.com/docker/compose/issues/6839
- all the calls to `local()` are run via `cmd` even when using PowerShell. I had to ajust some commands
- the `./run` of ngnix and php-fpm where broken. Bash was not available... Big WTF.
- I still have an issue with `"traefik.frontend.rule=Host:${PROJECT_HOSTNAMES}"` because I get an invalid TOML... Some investigations are needed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## Unreleased
+
+* Better Docker for Windows support
```
Comment (lyrixx) [inline CHANGELOG.md]: This should goes under the 2.0.0 as it's not released yet

File: fabfile.py [hunk 2]
```diff
@@ -208,22 +208,37 @@ def docker_compose_run(command_name, service="builder", user="app", no_deps=Fals
 
 def set_local_configuration():
     env.compose_files = ['docker-compose.yml']
-    env.user_id = int(local('id -u', capture=True))
-    env.root_dir = os.path.dirname(os.path.abspath(__file__))
+    env.dinghy = False
+    env.powerShell = False
```
Comment (lyrixx) [inline fabfile.py]: We are using snake_case for var name. Could you update `powerShell` to `power_shell`? Thanks

File: fabfile.py [hunk 3]
```diff
@@ -208,22 +208,37 @@ def docker_compose_run(command_name, service="builder", user="app", no_deps=Fals
...
 
+    env.root_dir = os.path.dirname(os.path.abspath(__file__))
+
     with quiet():
         try:
             docker_kernel = "%s" % local('docker version --format "{{.Server.KernelVersion}}"', capture=True)
         except:
             docker_kernel = ''
 
-    if platform == "linux" or platform == "linux2" or docker_kernel.endswith('linuxkit-aufs'):
```
Comment (pyrech) [inline fabfile.py]: Hum I wonder whether the check of `docker_kernel.endswith('linuxkit-aufs')` is still useful? Could this happen when platform is darwin (with docker for mac?)

Comment (damienalexandre) [reply fabfile.py]: Not a blocker for this PR.

Comment (damienalexandre) [reply fabfile.py]: Investigations points that the docker_kernel contains:

- `4.15.0-72-generic` on a Linux native Docker
- `4.14.98-boot2docker` on Dinghy
- `todo` on Docker For Mac
- `todo` on Docker For Windows

File: fabfile.py [hunk 4]
```diff
@@ -208,22 +208,39 @@ def docker_compose_run(command_name, service="builder", user="app", no_deps=Fals
...
+    if platform == "darwin" and docker_kernel.find('linuxkit') != -1:
         env.dinghy = True
+    elif platform in ["win32", "win64"]:  # TODO Find a better check, for PowerShell versus cmd
+        env.power_shell = True
+        # TODO this is a mess. Python can't set the vars correctly on PowerShell
+        # and local() always calls cmd.exe!
+        shellProjectName = local('echo %PROJECT_NAME%', capture=True)
+        if (shellProjectName != env.project_name):
+            domains = '`' + '`, `'.join([env.root_domain] + env.extra_domains) + '`'
+            print 'Did you run the env vars?'
```
Comment (lyrixx) [inline fabfile.py]:
```suggestion
            print 'You must set environment variables manually on windows:'
```

**Comment by damienalexandre** [comment]
I have no idea why the tests are broken now...

File: fabfile.py [hunk 5]
```diff
@@ -227,7 +227,7 @@ def set_local_configuration():
         shellProjectName = local('echo %PROJECT_NAME%', capture=True)
         if (shellProjectName != env.project_name):
             domains = '`' + '`, `'.join([env.root_domain] + env.extra_domains) + '`'
-            print 'Did you run the env vars?'
+            print 'You must manualy set environment variables on Windows:'
```
Comment (qboot) [inline fabfile.py]: typo: manualy -> manually

**Pull request merged**